lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -821,31 +909,31 @@
                         platform.raiseOptionValueOutOfRange();
                     }
 
-                    if (num === 0) {
+                    if (defaultCaseFirstCheck === 0) {
                         caseFirst = 'false';
-                    } else if (num === -1) {
+                    } else if (defaultCaseFirstCheck === -1) {
                         caseFirst = 'upper';
                     } else {
                         caseFirst = 'lower';
                     }
                 }
 
-                if (sensitivity === undefined) {
-                    sensitivity = "variant";
-                }
-
                 if (numeric === undefined) {
                     numeric = false;
                 }
 
                 // Set the options on the object
+                // REVIEW(jahorto): all of the enum-able keys get special fields for their enum
+                // so that we dont need to compute the key value from the enum in resolvedOptions()
                 collator.__matcher = matcher;
-                collator.__locale = resolvedLocaleInfo.localeWithoutSubtags;
-                collator.__localeForCompare = collationAugmentedLocale;
+                collator.__locale = finalLocale;
+                collator.__platformLocale = platformLocale;
                 collator.__usage = usage;
                 collator.__sensitivity = sensitivity;
+                collator.__sensitivityEnum = toEnum(CollatorSensitivity, sensitivity);
```

Comment:
>toEnum [](start = 45, length = 6)

Oh, toEnum is to convert a string to the numeric enum value. Maybe this should be given a more specific name?

Reply:
Any suggestions?